scripts/enforce_licence.py

```diff
@@ -280,38 +284,44 @@ def CheckLicenceForFile(file_name, licence, lang, diff, fix):
     return 0
 
   f = open(file_name)
+  rel_path = "./" + os.path.relpath(file_name)
   # + 1 to include the newline to have a complete line
   header_size = len(licence) + 1
+  file_name_line_count = licence.count('\n') + 2
   first_line = None
   if lang == PYTHON:
     first_line = f.readline()
     if not first_line.startswith('#!') and not first_line.startswith('# !'):
       # First line isn't a shebang, ignore it.
       f.seek(0, os.SEEK_SET)
       first_line = None
+    else:
+      file_name_line_count += 1
   # strip the trailing newline off as we don't actually want it
   header = f.read(header_size).rstrip('\n')
   file_name_line = f.readline()
   f.close()
   if header == licence:
     expected_line = TransformLine(os.path.basename(file_name), lang)
     if lang != JS and file_name_line.rstrip('\n') != expected_line:
-      print("File %s does not have a filename line after the licence; found "
-            "\"%s\" expected \"%s\"" %
-            (file_name, file_name_line.rstrip('\n'), expected_line))
+      print("error:file:%s:line %s: File does not have a filename line after "
+            "the licence; found \"%s\" expected \"%s\"" %
+            (rel_path, file_name_line_count, file_name_line.rstrip('\n'),
+             expected_line))
       return 1
     return 0
 
   if fix:
-    print('Fixing %s' % file_name)
+    print('notice:file:%s: Fixing file' % rel_path)
     if lang == PYTHON and first_line is not None:
       licence = first_line + licence
     ReplaceHeader(file_name, licence, lang)
     return 1
   else:
-    print("File %s does not start with \"%s...\"" % (
-        file_name,
-        licence.split('\n')[(0 if (lang == PYTHON) else 1)]))
+    print("error:file:%s:lines 1-%s: File does not start with or not exact "
+          "match of \"%s...\"" %
```

Comment:
I think this is slightly better English:
```suggestion
    print("error:file:%s:lines 1-%s: File does not start with, or not exact "
          "match of, \"%s...\"" %
```

Reply:
Fixed!